reconcile/deadmanssnitch.py

```diff
@@ -0,0 +1,226 @@
+        try:
+            full_secret_path = {
+                "path": snitch_secret_path,
+                "field": f"deadmanssnitch-{snitch.get_cluster_name()}-url",
+            }
+            snitch.vault_data = self.secret_reader.read(full_secret_path).strip()
+        except SecretNotFound:
+            snitch.vault_data = SECRET_NOT_FOUND
+        return snitch
+
+    def get_current_state(
+        self,
+        deadmanssnitch_api: DeadMansSnitchApi,
+        clusters: list[ClusterV1],
+        snitch_secret_path: str,
+    ) -> dict[str, Snitch]:
+        # current state includes for deadmanssnithch response and associated secret in vault
+        snitches = deadmanssnitch_api.get_snitches(tags=["app-sre"])
+        # create snitch_map only for  the desired clusters
+        snitches_with_cluster_mapping = {
+            snitch.get_cluster_name(): snitch for snitch in snitches
+        }
+        current_state = {
+            cluster.name: self.add_vault_data(snitch, snitch_secret_path)
+            for cluster in clusters
+            if (snitch := snitches_with_cluster_mapping.get(cluster.name))
+        }
+        return current_state
+
+    @staticmethod
+    def get_diff(
+        current_state: dict[str, Snitch], desired_state: list[ClusterV1]
+    ) -> list[DiffData]:
+        # get all the clusters which have `enableDeadMansSnitch: True`
+        desired_cluster_enabled_true = [
+            cluster.name for cluster in desired_state if cluster.enable_dead_mans_snitch
+        ]
+        # get all the clusters which have `enableDeadMansSnitch: False`
+        desired_cluster_enabled_false = [
+            cluster.name
+            for cluster in desired_state
+            if not cluster.enable_dead_mans_snitch
+        ]
+        create_snitches = [
+            DiffData(cluster_name=cluster, action=Action.create_snitch)
+            # add cluster to create set only in case they do not exist in current state
+            for cluster in desired_cluster_enabled_true - current_state.keys()
+        ]
+        delete_snitches = [
+            DiffData(
+                cluster_name=cluster,
+                action=Action.delete_snitch,
+                data=current_state[cluster].token,
+            )
+            # add to delete set only if they exists in current state and flag is set to false
+            for cluster in desired_cluster_enabled_false & current_state.keys()
+        ]
+        update_vault = [
+            DiffData(
+                cluster_name=cluster,
+                action=Action.update_vault,
+                data=current_state[cluster].check_in_url,
+            )
+            for cluster in desired_cluster_enabled_true & current_state.keys()
+            # add to update set only if they exists, flag is set to true but vault and snitch url mismatch
+            if current_state[cluster].vault_data
+            and current_state[cluster].vault_data != current_state[cluster].check_in_url
+        ]
```

Comment:
`current_state[cluster]` is used multiple times, can assign a var to it for reuse

```python
        [
            DiffData(
                cluster_name=cluster_name,
                action=Action.update_vault,
                data=cluster.check_in_url,
            )
            for cluster_name in desired_cluster_enabled_true & current_state.keys()
            if (cluster := current_state[cluster_name])
            and cluster.vault_data !=  cluster.check_in_url
        ]
```

Reply:
note `cluster` can be used in `data=cluster.check_in_url` too